src/main/java/org/folio/util/VertxUtils.java

```diff
@@ -19,11 +19,28 @@
    * Create a Pac4j {@link VertxWebContext} from {@link RoutingContext} with {@code null} SessionStore
    */
   public static VertxWebContext createWebContext(RoutingContext routingContext) {
-    return new VertxWebContext(routingContext, new DummySessionStore(routingContext.vertx(), routingContext.session()));
+    return new VertxWebContext2(routingContext, new DummySessionStore(routingContext.vertx(), routingContext.session()));
   }
 
   private VertxUtils() {}
 
+  private static class VertxWebContext2 extends VertxWebContext {
+
+    private RoutingContext routingContext;
+
+    public VertxWebContext2(RoutingContext routingContext, SessionStore<VertxWebContext> sessionStore) {
+      super(routingContext, sessionStore);
+      this.routingContext = routingContext;
+    }
+
+    @Override
+    public Optional<String> getRequestParameter(String name) {
+      Optional<String> param = super.getRequestParameter(name);
+      return param.isPresent() ? param : Optional.ofNullable(routingContext.request().formAttributes().get(name));
+    }
+
+  }
+
   public static class DummySessionStore implements SessionStore<VertxWebContext>  {
```

Comment:
If this is really an issue in vertx-pac4j , we should fix it there.

We are maintaining it at the moment !! https://github.com/folio-org/vertx-pac4j